### PR TITLE
update to 7.1 and refactor

### PR DESCRIPTION
**Pull request by vincentsarago**
This PR does:
- update to eoapi-cdk 7.1
- refactor and simplify the app.py
- update the Config to enable `.env` files 
- add `.env.example` and `config.yml.example` files
- update dockerfiles versions 
-

### CODE REVIEW

**Comment by vincentsarago** [comment]
deployed and ✅

File: infrastructure/app.py [hunk 1]
```diff
@@ -1,17 +1,393 @@
...
+        self.export_value(
+            self.vpc.select_subnets(subnet_type=aws_ec2.SubnetType.PUBLIC)
+            .subnets[0]
+            .subnet_id
+        )
+        self.export_value(
+            self.vpc.select_subnets(subnet_type=aws_ec2.SubnetType.PUBLIC)
+            .subnets[1]
+            .subnet_id
+        )
```
Comment (vincentsarago) [inline infrastructure/app.py:78]: move VpcSTAC in app.py

File: infrastructure/app.py [hunk 2]
```diff
@@ -1,17 +1,393 @@
...
+                "description": f"{app_config.stage} STAC API",
+            },
+            db=pgstac_db.db,
+            db_secret=pgstac_db.pgstac_secret,
+            # If the db is not in the public subnet then we need to put
+            # the lambda within the VPC
+            vpc=vpc if not app_config.public_db_subnet else None,
+            subnet_selection=aws_ec2.SubnetSelection(
+                subnet_type=aws_ec2.SubnetType.PRIVATE_WITH_EGRESS
+            )
```
Comment (vincentsarago) [inline infrastructure/app.py:135]: if the database is in a private subnet we need to put the lambdas in the private subnet as well

File: infrastructure/config.py [hunk 3]
```diff
@@ -110,12 +113,14 @@ class AppConfig(BaseSettings):
         default=None,
     )
 
-    @pydantic.field_validator("tags")
-    def default_tags(cls, v, info: FieldValidationInfo):
+    model_config = SettingsConfigDict(env_file=".env", yaml_file="config.yaml")
```
Comment (vincentsarago) [inline infrastructure/config.py:116]: Pydantic-settings now supports yaml

File: infrastructure/config.py [hunk 4]
```diff
@@ -138,43 +139,38 @@ def validate_stac_browser_version(self) -> "AppConfig":
...
+        dotenv_settings: PydanticBaseSettingsSource,
+        file_secret_settings: PydanticBaseSettingsSource,
+    ) -> Tuple[PydanticBaseSettingsSource, ...]:
+        return (
+            init_settings,
+            env_settings,
+            dotenv_settings,
+            file_secret_settings,
+            YamlConfigSettingsSource(settings_cls),
+        )
```
Comment (vincentsarago) [inline infrastructure/config.py:176]: yaml support comes in last priority

File: config.yaml.example [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+# STAC Browser Options
+stac_browser_version: '0.1.0'
+
+# API Gateway Options
+acm_certificate_arn: 'arn:...'
+stac_api_custom_domain: 'https://stac.api'
+raster_api_custom_domain: 'https://raster.api'
+vector_api_custom_domain: 'https://vector.api'
+stac_ingestor_api_custom_domain: 'https://ingestor.api'
```
Comment (vincentsarago) [inline config.yaml.example:36]: closes #23

**Review by emileten — APPROVED** [review]
Looks good to me :) Thanks for the cleanup

As long as it deploys and is functional 👍

**Pull request merged**